packages/backend-common/src/service/types.ts

```diff
@@ -85,6 +85,17 @@ export type ServiceBuilder = {
    */
   addRouter(root: string, router: Router | RequestHandler): ServiceBuilder;
 
+  /**
+   * Set the request logging handler
+   *
+   * If no handler is given the default one is used
+   *
+   * @param requestLoggingHandler a factory function that given a logger returns an handler
+   */
+  setRequestLoggingHandler(
+    requestLoggingHandler: (logger?: Logger) => RequestHandler,
```

Comment:
Why is the logger optional argument? Just to make typescript happy if you don't want to use the input parameter?

Reply:
I just got the signature from the original function:  https://github.com/backstage/backstage/blob/01647758ed97044bac75c7b786512bdb48f27f3a/packages/backend-common/src/middleware/requestLoggingHandler.ts#L28

`export function requestLoggingHandler(logger?: Logger): RequestHandler {`

As the function function is going to be called from `ServiceBuilderImpl.start` the logger is going to be always defined, if you prefer I can make it non optional.

Reply:
Ah yeah - sorry misread that, yeah maybe let's create a seperate type for it that we can use as the public API rather than using `typeof`

Reply:
I created a `RequestLoggingHandlerFactory` type to be used in the public api of the `ServiceBuilder.setRequestLoggingHandler`